persistence/sql/persister_recovery.go

```diff
@@ -75,12 +74,10 @@ func (p *Persister) UseRecoveryToken(ctx context.Context, token string) (*link.R
 		}
 
 		var ra identity.RecoveryAddress
-		if err := tx.Where("id = ? AND nid = ?", rt.RecoveryAddressID, nid).First(&ra); err != nil {
-			return sqlcon.HandleError(err)
+		if err := tx.Where("id = ? AND nid = ?", rt.RecoveryAddressID, nid).First(&ra); err == nil {
```

Comment:
Any error would be ignored here. Please return them appropriately. You can use `errors.Is(sql.ErrNoRows)` to detect if the row could not be found